docs/developmentSetup.md

```diff
@@ -7,22 +7,96 @@ To begin we will need to set up the code base on your own machine. To do this we
+ **Do not commit MAILGUN_API_KEY or share it outside c0d3.**
+```
+   MAILGUN_API_KEY=123abc #ask for valid key in chat
+   MATTERMOST_ACCESS_TOKEN=123abc #you will generate it yourself
+   CHAT_URL=http://localhost:8000/api/v4 #mattermost api 
+   CLIENT_URL=http://localhost:4000 #used in registration email
+   #postgres user, database name, password, host and port
+   DB_USER=c0d3_admin
+   DB_NAME=c0d3
+   DB_PW=admin
+   DB_HOST=localhost
+   DB_PORT=7000
+```
+6. Build docker images. Go to docker directory and type `docker-compose build`. It will take some time for docker to pull and install everything. In the end you will have one container with a postgres database and one container for mattermost. 
+
+7. Prepare volume directories. Volumes are mappings from containers to your local machine.
+```
+mkdir -pv ./volumes/app/mattermost/{data,logs,config,plugins,client-plugins}
+sudo chown -R 2000:2000 ./volumes/app/mattermost/
+```
+8. Now you can start docker `docker-compose up` (add -d flag to start in detached mode). To stop all containers `docker-compose down`.
+
+9. Configure your local mattermost server. When you open [http://localhost:8080](http://localhost:8000), you will be greeted with the mattermost login page. Enter any email (this setup won't have an SMTP server, so no actual mails will be sent). First user automatically becomes the server admin. Create new team and enable use of personal access tokens *System Console > Integrations > Custom Integrations*.
+
+10. Generate a personal access token. Close system console, go to team chat (town square) and click on three horizontal bars near your name *Account settings > Security* , then `generate personal access token`. This value will be used as MATTERMOST_ACCESS_TOKEN.
+
+11. Keep in mind that by default mattermost requires very secure passwords: at least 10 characters, at least one uppercase letter, one number and one special symbol. Either use such passwords or edit mattermost config at *docker/docker/volumes/app/mattermost/config/config.json* PasswordSettings and restart the server.
+
+12. Allow new users to join your team. As admin go to *Team settings/Allow any user with an account on this server to join this team*. 
+
+13. Start c0d3 app `PORT=4000 yarn dev`. If everything went right you will see the landing page. Login with `admin:password` or register new user.
+
+
+To submit challenges:
+1. Logout `c0d3 logout`.
+2. Login to your local server `c0d3 login --url http://localhost:4000/api/graphql`.
+3. Submit your code `c0d3 submit --url http://localhost:4000/api/graphql`
+It should be visible on your local setup.
+
+To connect to postgres database inside container:
+1. Check your current active docker containers `docker ps`. Notice docker_postgres container id (first few characters are enough).
+2. Open bash console inside postgres container `docker exec -it *postgres id* /bin/bash`.
```

Comment:
The container could be named, then the container name can be used instead of the id.

Use [container_name](https://docs.docker.com/compose/compose-file/compose-file-v3/#container_name) in the compose file for that.